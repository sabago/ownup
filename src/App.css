.App {
  text-align: center;
  width: 100%;
  min-width: 400px;
  margin: 1% auto;
  padding-left: 20%;
}
