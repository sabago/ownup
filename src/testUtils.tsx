import React from 'react';
import { render as rtlRender } from '@testing-library/react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
// Import your own reducer
// import reducer from '../reducer'
import reduxThunkReducer from './reduxLogic/apiReducer';

 
const testRender = (ui: any, {
    // initialState,
    // store = createStore(reducer, initialState),
    store = createStore(reduxThunkReducer),
    ...renderOptions
  } = {}
) =>{
  const Wrapper = (children: React.ReactNode) => {
    return <Provider store={store}>{ui}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// {homes.map(home => <div>{home.name}</div>)}

// re-export everything
export * from '@testing-library/react';
// override render method
export { testRender };

