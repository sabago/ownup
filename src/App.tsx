import * as React from 'react';
import {useState, useEffect} from 'react';
import './App.css';

interface IProps {
}

interface IState {
  playOrPause?: string;
}

let data:any;

const App: React.FC<IProps> = (props:IProps, state:IState) => {
  const [loanSize, setLoanSize] = useState(450000);
  const [creditScore, setCreditScore] = useState(680);
  const [propertyType, setPropertyType] = useState("SingleFamily");
  const [occupancy, setOccupancy] = useState("Primary");

  useEffect(() => {
    const API_KEY = process.env.REACT_APP_WEATHER_API_KEY;
    const API_URL= `https://ss6b2ke2ca.execute-api.us-east-1.amazonaws.com/Prod/quotes/?apiKey=${API_KEY}`;
    const fetch_url = new URL(API_URL);
    fetch_url.search = new URLSearchParams({"loanSize":`${loanSize}`, "creditScore":`${creditScore}`, "propertyType": `${propertyType}`, "occupancy":`${occupancy}`}).toString();

    const fetchData = async () => {
    try {
      const response = await fetch(fetch_url.href, {
        headers: {
            'Authorization': `OU-AUTH ${API_KEY}`
            }
      });
      const data = await response.text();
      const dataObj = JSON.parse(data);

      //delete  when done with table     
      console.log(dataObj["rateQuotes"]);

    } catch(error) {
      console.log(error);
    }
  }

  fetchData();
  },[loanSize , creditScore, propertyType, occupancy]);

  return (
    <div className="App">
      {data}
    </div>
  );
}
  

export default App;
