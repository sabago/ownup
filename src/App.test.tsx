import React from 'react';
import App from './App';
import { testRender } from './testUtils';



describe("when the user opens the browser", () => {
  it("renders App component in initial state", () => {
    const { asFragment } = testRender(<App/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders App component with the Search Contol component", ()=>{
    const { getByTestId } = testRender(<App />);
    expect(getByTestId("search-control")).toBeInTheDocument();
  });

  it("renders App component with the Table component", ()=>{
    const { getByTestId } = testRender(<App />);
    expect(getByTestId("quote-table")).toBeInTheDocument();
  });
})

//Add E2E test for when user clicks the button in the search control