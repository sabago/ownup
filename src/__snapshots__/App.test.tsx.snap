// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the user opens the browser renders App component in initial state 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      data-cy="search-control"
      data-testid="search-control"
    >
      <form>
        <div
          class="search"
        >
          <span>
            <label
              id="loan-size"
            >
              Loan Size
            </label>
            <input
              aria-labelledby="loan-size"
              data-cy="loan-size"
              data-testid="loan-size"
              placeholder="450000"
              type="number"
              value="450000"
            />
          </span>
          <span>
            <label
              id="property-type"
            >
              Property Type
            </label>
            <select
              aria-labelledby="property-type"
              data-cy="property-type"
              id="property-type"
            >
              <option
                value="SingleFamily"
              >
                Single Family
              </option>
              <option
                value="Condo"
              >
                Condo
              </option>
              <option
                value="Townhouse"
              >
                Townhouse
              </option>
              <option
                value="MultiFamily"
              >
                Multi Family
              </option>
            </select>
          </span>
        </div>
        <br />
        <div
          class="search"
        >
          <span>
            <label
              id="credit-score"
            >
               Credit Score
            </label>
            <input
              aria-labelledby="credit-score"
              data-cy="credit-score"
              data-testid="credit-score"
              placeholder="680"
              type="number"
              value="680"
            />
          </span>
          <span>
            <label
              id="occupancy"
            >
              Occupancy
            </label>
            <select
              aria-labelledby="occupancy"
              data-cy="occupancy"
            >
              <option
                value="Primary"
              >
                Primary Residence
              </option>
              <option
                value="Secondary"
              >
                Secondary Residence
              </option>
              <option
                value="Investment"
              >
                Investment
              </option>
            </select>
          </span>
        </div>
        <button
          class="quote-button"
          data-cy="quote-button"
        >
          Quote Rates
        </button>
      </form>
    </div>
    <div
      data-cy="quote-table"
      data-testid="quote-table"
    >
      <div
        class="center-loader"
        data-cy="loader"
        data-testid="loader"
      >
        <h1>
           Press the button to see your rates!
        </h1>
        <div
          aria-busy="true"
          class=""
        >
          <svg
            aria-label="audio-loading"
            height="80"
            viewBox="0 0 38 38"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <lineargradient
                id="a"
                x1="8.042%"
                x2="65.682%"
                y1="0%"
                y2="23.865%"
              >
                <stop
                  offset="0%"
                  stop-color="#265b5f"
                  stop-opacity="0"
                />
                <stop
                  offset="63.146%"
                  stop-color="#265b5f"
                  stop-opacity=".631"
                />
                <stop
                  offset="100%"
                  stop-color="#265b5f"
                />
              </lineargradient>
            </defs>
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                transform="translate(1 1)"
              >
                <path
                  d="M36 18c0-9.94-8.06-18-18-18"
                  id="Oval-2"
                  stroke="#265b5f"
                  stroke-width="2"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="0.9s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </path>
                <circle
                  cx="36"
                  cy="18"
                  fill="#fff"
                  r="1"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="0.9s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </circle>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
